File: applicatie/src/main/java/com/vdzon/administratie/gebruiker/GebruikerService.java
package com.vdzon.administratie.gebruiker;

import com.fasterxml.jackson.databind.ObjectMapper;
import com.vdzon.administratie.auth.SessionHelper;
import com.vdzon.administratie.crud.UserCrud;
import com.vdzon.administratie.dto.FactuurDto;
import com.vdzon.administratie.dto.GebruikerDto;
import com.vdzon.administratie.model.Factuur;
import com.vdzon.administratie.model.Gebruiker;
import com.vdzon.administratie.util.SingleAnswer;
import spark.Request;
import spark.Response;

import javax.inject.Inject;
import java.util.List;
import java.util.stream.Collector;
import java.util.stream.Collectors;

public class GebruikerService {

    @Inject
    UserCrud crudService;

    protected Object postGebruiker(Request req, Response res) throws Exception {
        try {
            String uuid = SessionHelper.getAuthenticatedUserUuid(req);
            Gebruiker gebruiker = crudService.getGebruiker(uuid);
            if (gebruiker == null) {
                res.status(404);
                return new SingleAnswer("not found");
            }
            String nieuweGebruikerJson = req.body();
            ObjectMapper mapper = new ObjectMapper();
            GebruikerDto nieuweGebruikerDto = mapper.readValue(nieuweGebruikerJson, GebruikerDto.class);
            Gebruiker originalGebruiker = crudService.getGebruiker(nieuweGebruikerDto.getUuid());
            if (originalGebruiker == null){
                originalGebruiker = new Gebruiker();
            }
            Gebruiker updatedGebruiker  = nieuweGebruikerDto.cloneGebruikerAndCopyDtoFields(originalGebruiker);
            crudService.updateGebruiker(updatedGebruiker );
            return new SingleAnswer("ok");
        } catch (Exception e) {
            e.printStackTrace();
            throw e;
        }
    }

    protected Object removeGebruiker(Request req, Response res) throws Exception {
        String uuid = SessionHelper.getAuthenticatedUserUuid(req);
        Gebruiker gebruiker = crudService.getGebruiker(uuid);
        if (gebruiker == null) {
            res.status(404);
            return new SingleAnswer("not found");
        }
        String gebruikerUuid = req.params(":uuid");
        if ("undefined".equals(gebruikerUuid)) {
            gebruikerUuid = null;
        }
        crudService.deleteGebruiker(gebruikerUuid);;
        return new SingleAnswer("ok");
    }

    protected Object updatePassword(Request req, Response res) throws Exception {
        String uuid = SessionHelper.getAuthenticatedUserUuid(req);
        Gebruiker gebruiker = crudService.getGebruiker(uuid);
        if (gebruiker == null) {
            res.status(404);
            return new SingleAnswer("not found");
        }
        String gebruikerUuid = req.params(":uuid");
        String newPassword = req.params(":newPassword");
        Gebruiker gebruikerToChange = crudService.getGebruiker(gebruikerUuid);
        gebruikerToChange.setPassword(newPassword);
        crudService.updateGebruiker(gebruikerToChange);
        return new SingleAnswer("ok");
    }

}

File: rest/impl/src/main/java/com/vdzon/administratie/rest/gebruiker/GebruikerService.kt
package com.vdzon.administratie.rest.gebruiker

import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
import com.vdzon.administratie.authenticatie.AuthenticationService
import com.vdzon.administratie.database.UserDao
import com.vdzon.administratie.dto.GebruikerDto
import com.vdzon.administratie.model.Administratie
import com.vdzon.administratie.model.Gebruiker
import com.vdzon.administratie.util.SingleAnswer
import spark.Request
import spark.Response
import java.util.*
import javax.inject.Inject

class GebruikerService {

    @Inject
    lateinit internal var daoService: UserDao

    @Inject
    lateinit internal var athenticationService: AuthenticationService

    @Throws(Exception::class)
    fun postGebruiker(req: Request, res: Response): Any {
        val gebruiker = athenticationService.getGebruikerOrThowForbiddenException(req, res)
        val nieuweGebruikerJson = req.body()
        val mapper = jacksonObjectMapper()
        val nieuweGebruikerDto = mapper.readValue(nieuweGebruikerJson, GebruikerDto::class.java)
        var originalGebruiker: Gebruiker? = daoService!!.getGebruiker(nieuweGebruikerDto.uuid)
        if (originalGebruiker == null) {
            originalGebruiker = Gebruiker("", "", "", false, ArrayList<Administratie>())
        }
        val updatedGebruiker = nieuweGebruikerDto.cloneGebruikerWithDtoFields(originalGebruiker)
        daoService!!.updateGebruiker(updatedGebruiker)
        return SingleAnswer("ok")
    }

    @Throws(Exception::class)
    fun removeGebruiker(req: Request, res: Response): Any {
        val gebruiker = athenticationService.getGebruikerOrThowForbiddenException(req, res)
        var gebruikerUuid: String? = req.params(":uuid")
        if ("undefined" == gebruikerUuid) {
            gebruikerUuid = null
        }
        daoService!!.deleteGebruiker(gebruikerUuid)
        return SingleAnswer("ok")
    }

}
